src/components/Dialog/index.tsx

```diff
@@ -96,28 +96,49 @@ export function Outer({
    */
   const isOpen = openIndex > -1
 
+  const callQueuedCallbacks = React.useCallback(() => {
+    for (const cb of closeCallbacks.current) {
+      try {
+        cb()
+      } catch (e: any) {
+        logger.error('Error running close callback', e)
+      }
+    }
+
+    closeCallbacks.current = []
+  }, [])
+
   const open = React.useCallback<DialogControlProps['open']>(
     ({index} = {}) => {
       setDialogIsOpen(control.id, true)
       // can be set to any index of `snapPoints`, but `0` is the first i.e. "open"
       setOpenIndex(index || 0)
+
+      callQueuedCallbacks()
```

Comment:
Do we want to call all close callbacks here, or maybe just reset the an empty array on re-open?

Reply:
I think calling all of the old ones is the best move. I.e:

- Press save
- Close animation begins
- Somehow the dialog gets opened again before the animation ends
- ???

I think i'd expect the first thing I did to still complete personally.